references/references.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2023 Nephio.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package references
+
+
+// NamespaceReference is used to reference a particular krm intstance
+type NamespaceReference struct {
```

Comment:
We do have this already:

https://pkg.go.dev/k8s.io/api@v0.26.3/core/v1#ObjectReference

Is that too general?

Reply:
I am fine using this but we don't need most of these objects